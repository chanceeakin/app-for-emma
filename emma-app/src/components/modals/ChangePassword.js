// @flow
import React, { Component } from 'react'
import {
  Modal,
  Text,
  TouchableHighlight,
  View,
  StyleSheet,
  TouchableOpacity,
  Keyboard,
  TextInput,
} from 'react-native'
import GradientWrapper from './../wrappers/GradientWrapper'
import {
  forms,
  colors,
  button,
  modal as modalStyle,
  pageLayout,
} from './../../styles'
import type { SettingsAction } from './../../actions/Settings.js.flow'

const {
  bigButton,
  bigButtonText,
  settingsButton,
  settingsButtonText,
  errorButton,
  errorText,
  backButton,
} = button
const { modal, modalTitle } = modalStyle
const { container } = pageLayout
const { input } = forms

const styles = StyleSheet.create({
  container,
  input,
  modal,
  modalTitle,
  bigButton,
  bigButtonText,
  settingsButton,
  settingsButtonText,
  errorButton,
  errorText,
  backButton,
})

type Props = {
  updateOldPasswordPatchTextField: SettingsAction,
  updateNewPassword1PatchTextField: SettingsAction,
  updateNewPassword2PatchTextField: SettingsAction,
  updatedOldPassword: string,
  updatedNewPassword1: string,
  updatedNewPassword2: string,
  isPasswordModalShown: boolean,
  togglePasswordModal: SettingsAction,
};
export default class ModalComponent extends Component<Props> {
  render() {
    const {
      updateOldPasswordPatchTextField,
      updatedOldPassword,
      updateNewPassword1PatchTextField,
      updateNewPassword2PatchTextField,
      updatedNewPassword1,
      updatedNewPassword2,
      togglePasswordModal,
      isPasswordModalShown,
    } = this.props
    return (
      <View style={styles.container}>
        <Modal
          animationType="fade"
          transparent={false}
          visible={isPasswordModalShown}
          onRequestClose={() => {
            alert('Modal has been closed.')
          }}
        >
          <GradientWrapper color="orange" style={styles.modal}>
            <Text style={styles.modalTitle}>Change Password</Text>
            <View>
              <TextInput
                style={styles.input}
                onChangeText={text => updateOldPasswordPatchTextField(text)}
                value={updatedOldPassword}
                secureTextEntry
                placeholder="Enter Old Password"
                placeholderTextColor={colors.inkBlue}
                autoCapitalize="none"
                onBlur={Keyboard.dismiss}
              />
              <TextInput
                style={styles.input}
                onChangeText={text => updateNewPassword1PatchTextField(text)}
                value={updatedNewPassword1}
                secureTextEntry
                placeholder="Enter New Password"
                placeholderTextColor={colors.inkBlue}
                autoCapitalize="none"
                onBlur={Keyboard.dismiss}
              />
              <TextInput
                style={styles.input}
                onChangeText={text => updateNewPassword2PatchTextField(text)}
                value={updatedNewPassword2}
                secureTextEntry
                placeholder="Enter New Password Again"
                placeholderTextColor={colors.inkBlue}
                autoCapitalize="none"
                onBlur={Keyboard.dismiss}
              />
              <TouchableOpacity style={styles.bigButton}>
                <Text style={styles.bigButtonText}>Change Password</Text>
              </TouchableOpacity>
            </View>

            <TouchableHighlight
              style={styles.backButton}
              onPress={togglePasswordModal}
            >
              <Text>Back</Text>
            </TouchableHighlight>
          </GradientWrapper>
        </Modal>

        <TouchableHighlight
          style={styles.settingsButton}
          onPress={togglePasswordModal}
        >
          <Text style={styles.settingsButtonText}>Change Password</Text>
        </TouchableHighlight>
      </View>
    )
  }
}
