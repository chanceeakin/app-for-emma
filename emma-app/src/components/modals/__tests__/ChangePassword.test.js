import React from 'react'
import ChangePassword from '../ChangePassword'

import renderer from 'react-test-renderer'

const props = {
  updatedEmail: 'email',
  updateEmailPatchTextField: jest.fn(),
  isPatchingEmailError: false,
  changeEmail: jest.fn(),
  isEmailModalShown: false,
  toggleEmailModal: jest.fn(),
}

jest.mock('TouchableOpacity', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableOpacity')
})

jest.mock('TouchableHighlight', () => {
  const mockComponent = require('jest-react-native')
  return mockComponent('TouchableHighlight')
})

describe('<ChangePassword />', () => {
  afterAll = () => {
    jest.mockRestore()
  }
  it('renders correctly', () => {
    const tree = renderer.create(<ChangePassword {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
