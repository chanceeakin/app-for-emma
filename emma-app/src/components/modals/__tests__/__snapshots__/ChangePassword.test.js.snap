// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePassword /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "space-between",
    }
  }
>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={false}
    visible={true}
  >
    <ExponentLinearGradient
      colors={
        Array [
          4294960596,
          4293904322,
        ]
      }
      endPoint={
        Array [
          1,
          1,
        ]
      }
      location={
        Array [
          0.25,
          0.4,
          1,
        ]
      }
      locations={null}
      startPoint={
        Array [
          0,
          0,
        ]
      }
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "space-between",
          "paddingTop": 30,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 45,
            "textAlign": "center",
          }
        }
      >
        Change Password
      </Text>
      <View>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Enter Old Password"
          placeholderTextColor="#0D1B2A"
          secureTextEntry={true}
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "gray",
              "color": "#114B5F",
              "height": 40,
              "minWidth": 200,
            }
          }
          value={undefined}
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Enter New Password"
          placeholderTextColor="#0D1B2A"
          secureTextEntry={true}
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "gray",
              "color": "#114B5F",
              "height": 40,
              "minWidth": 200,
            }
          }
          value={undefined}
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          placeholder="Enter New Password Again"
          placeholderTextColor="#0D1B2A"
          secureTextEntry={true}
          style={
            Object {
              "borderBottomWidth": 1,
              "borderColor": "gray",
              "color": "#114B5F",
              "height": 40,
              "minWidth": 200,
            }
          }
          value={undefined}
        />
        <TouchableOpacity
          activeOpacity={0.2}
          style={
            Object {
              "borderColor": "#0D1B2A",
              "borderRadius": 5,
              "borderStyle": "solid",
              "borderWidth": 1,
              "marginTop": 20,
              "paddingHorizontal": 40,
              "paddingVertical": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#0D1B2A",
                "fontSize": 23,
                "fontWeight": "600",
              }
            }
          >
            Change Password
          </Text>
        </TouchableOpacity>
      </View>
      <TouchableHighlight
        activeOpacity={0.85}
        delayPressOut={100}
        onPress={undefined}
        style={
          Object {
            "padding": 15,
          }
        }
        underlayColor="black"
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Back
        </Text>
      </TouchableHighlight>
    </ExponentLinearGradient>
  </Modal>
  <TouchableHighlight
    activeOpacity={0.85}
    delayPressOut={100}
    onPress={undefined}
    style={
      Object {
        "backgroundColor": "#D37482",
        "borderRadius": 5,
        "marginTop": 20,
        "paddingHorizontal": 30,
        "paddingVertical": 10,
      }
    }
    underlayColor="black"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#fafafa",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      Change Password
    </Text>
  </TouchableHighlight>
</View>
`;
